pkg/client/vcwallet/client.go

```diff
@@ -9,25 +9,139 @@ package vcwallet
 import (
 	"encoding/json"
 	"fmt"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
 )
 
 // provider contains dependencies for the verifiable credential wallet client
 // and is typically created by using aries.Context().
 type provider interface {
-	// TODO to be added #2433
+	StorageProvider() storage.Provider
+}
+
+// kmsOpts contains options for creating verifiable credential wallet client.
+type kmsOpts struct {
+	// local kms options
+	secretLockSvc secretlock.Service
+	passphrase    string
+
+	// remote(web) kms options
+	keyServerURL string
+}
+
+// KeyManagerOptions is option for verifiable credential wallet client key manager.
+type KeyManagerOptions func(opts *kmsOpts)
+
+// WithSecretLockService option, when provided then wallet client will use local kms for key operations.
+func WithSecretLockService(svc secretlock.Service) KeyManagerOptions {
+	return func(opts *kmsOpts) {
+		opts.secretLockSvc = svc
+	}
+}
+
+// WithPassphrase option to provide passphrase for local kms for key operations.
+func WithPassphrase(passphrase string) KeyManagerOptions {
+	return func(opts *kmsOpts) {
+		opts.passphrase = passphrase
+	}
+}
+
+// WithKeyServerURL option, when provided then wallet client will use remote kms for key operations.
+// This option will be ignore if provided with 'WithSecretLockService' option.
+func WithKeyServerURL(url string) KeyManagerOptions {
+	return func(opts *kmsOpts) {
+		opts.keyServerURL = url
+	}
 }
 
 // Client enable access to verifiable credential wallet features.
 type Client struct {
 	// ID of wallet content owner
 	userID string
+
+	// wallet profile
+	profile *profile
+
+	// storage provider
+	storeProvider storage.Provider
+}
+
+// Get returns new verifiable credential wallet client for given user.
+// returns error if wallet profile is not found.
+// To create a new client with new wallet profile, use `New()`.
+func Get(userID string, ctx provider) (*Client, error) {
+	store, err := newProfileStore(ctx.StorageProvider())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get store to fetch VC wallet profile info: %w", err)
+	}
+
+	profile, err := store.get(userID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get VC wallet profile: %w", err)
+	}
+
+	return &Client{userID: userID, profile: profile, storeProvider: ctx.StorageProvider()}, nil
+}
+
+// Create creates and returns new verifiable credential wallet client by creating new wallet profile for given user.
+// returns error if wallet profile is already created.
+// Use `Get()` for already created VC wallet profile.
+// Use `Update()` for replacing an already created VC wallet profile.
+func Create(userID string, ctx provider, options ...KeyManagerOptions) (*Client, error) {
```

Comment:
It might be a bit confusing to have New/Get,Create,Update all return clients.

I wonder if Create/Update should be CreateProfile and UpdateProfile and they don't return clients?

Reply:
note: I'm also not entirely sure if we should be mixing `Create`/`Update` with returning a `Client`.

Reply:
If these functions names / roles are more clear, it is easier to use the normal `New` method name for the client.

Reply:
Looks good to me, 
call `CreateProfile()` once and use `New()` whenever you want client.

fixed: 

- `CreateProfile/UpdateProfile` with returning error
- `New` returning client